lib/internal/bootstrap/node.js

```diff
@@ -72,75 +70,15 @@ process._exiting = false;
 // process.config is serialized config.gypi
 const nativeModule = internalBinding('native_module');
 
-// TODO(@jasnell): Once this has gone through one full major
-// release cycle, remove the Proxy and setter and update the
-// getter to either return a read-only object or always return
-// a freshly parsed version of nativeModule.config.
-
-const deprecationHandler = {
-  warned: false,
-  message: 'Setting process.config is deprecated. ' +
-           'In the future the property will be read-only.',
-  code: 'DEP0150',
-  maybeWarn() {
-    if (!this.warned) {
-      process.emitWarning(this.message, {
-        type: 'DeprecationWarning',
-        code: this.code
-      });
-      this.warned = true;
-    }
-  },
-
-  defineProperty(target, key, descriptor) {
-    this.maybeWarn();
-    return ObjectDefineProperty(target, key, descriptor);
-  },
-
-  deleteProperty(target, key) {
-    this.maybeWarn();
-    delete target[key];
-  },
-
-  preventExtensions(target) {
-    this.maybeWarn();
-    return ObjectPreventExtensions(target);
-  },
-
-  set(target, key, value) {
-    this.maybeWarn();
-    return ReflectSet(target, key, value);
-  },
-
-  get(target, key, receiver) {
-    const val = ReflectGet(target, key, receiver);
-    if (val != null && typeof val === 'object') {
-      // eslint-disable-next-line node-core/prefer-primordials
-      return new Proxy(val, deprecationHandler);
-    }
-    return val;
-  },
-
-  setPrototypeOf(target, proto) {
-    this.maybeWarn();
-    return ObjectSetPrototypeOf(target, proto);
-  }
-};
-
-// eslint-disable-next-line node-core/prefer-primordials
-let processConfig = new Proxy(
-  JSONParse(nativeModule.config),
-  deprecationHandler);
+const processConfig = JSONParse(nativeModule.config, (_key, value) => {
```

Comment:
I don't think it should be required: the `reviver` function will be called on the "root" object too, so no need for that. Although we can still do this if you feel like it will be more explicit that way, doesn't seem like there is any harm in doing this, what do you think?

Reply:
If that's not required, that's fine, let's add a comment though.

Reply:
Added a comment